### DISCUSSION TITLE
curl logs "400 Bad Request" while attempting to download using FOUNDRY_RELEASE_URL

### DISCUSSION THREAD

**Discussion by archer31** [felddy/foundryvtt-docker#132] [Q&A] [answered]
Hi guys I am having issues getting the container to download the release using the `FOUNDRY_RELEASE_URL` ENV variable.
if I have it set to the full URI then curl complains that it cannot resolve the host `https`, and if I remove the `https://` from the front of the URI then I get a 400 bad request.

my URI looks something like `https://foundryvtt.s3.amazonaws.com/releases/0.7.9/foundryvtt-0.7.9.zip?AWSAccessKeyId=<access id>&Signature=<signature id>&Expires=<expiration>` copied directly from the foundry site.

logs:
```
Entrypoint | 2020-12-21 22:44:07 | [info] Starting felddy/foundryvtt container v0.7.9,
Entrypoint | 2020-12-21 22:44:07 | [info] No Foundry Virtual Tabletop installation detected.,
Entrypoint | 2020-12-21 22:44:07 | [info] Using FOUNDRY_RELEASE_URL to download release.
Entrypoint | 2020-12-21 22:43:06 | [info] Downloading Foundry Virtual Tabletop release.,
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current,
                                 Dload  Upload   Total   Spent    Left  Speed,
  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0,
curl: (22) The requested URL returned error: 400 Bad Request
```

username and password auth sadly wasn't working for me even though I could login to my browser fine.

running tag: release

**Comment by felddy** [comment, marked as answer]
Here are a few ideas:
 - It could be a quoting issue.
 - Make sure to use the pre-signed S3 URL within 5 minutes.  It expires.
 - Try turning on 'CONTAINER_VERBOSE` and get more detailed logs.
 
If none of these helps, you download the zip a different way and place it in a cache directory.  Then set `CONTAINER_CACHE`.  See: https://github.com/felddy/foundryvtt-docker#pre-cached-distribution-variable
